lib/pacemaker/pcmk_graph_producer.c

```diff
@@ -636,10 +636,10 @@ should_add_input_to_graph(const pe_action_t *action, pe_action_wrapper_t *input)
                   input->action->uuid, input->action->id);
         return false;
 
-    } else if (pcmk_is_set(input->type, pe_order_implies_first_migratable)
+    } else if (pcmk_is_set(input->type, pcmk_ord_unmigratable_then_blocks)
                && !pcmk_is_set(input->action->flags, pcmk_action_runnable)) {
         crm_trace("Ignoring %s (%d) input %s (%d): "
-                  "implies input migratable but input unrunnable",
+                  "input blocked if 'then' unmigratable",
```

Comment:
By the time we get here, the relationships have already been applied, so it appears to be checking for the result of that application rather than the conditions. I'm sure that means there are other conditions that pass the test, but it's probably sufficient for determining whether to graph the input.

Reply:
That doesn't leave this section any less confusing, but that works for me if the behavior is correct.